### DISCUSSION TITLE
Creating edge_index for mesh graph

### DISCUSSION THREAD

**Discussion by atale480** [pyg-team/pytorch_geometric#5239] [Q&A] [answered]
I'm creating own dataset. I have adjacency list representation of graph but PyTorch_Gometric need  Graph connectivity in COO format for edge_index. I'm having 3D Mesh in off file format and off according to off file 

> mesh has 'x' number of edges. 

I converted mesh to adjacency list format. Now how can I convert adjacency list to edge_index ?
I did it with following piece of code.

> import open3d as o3d
> import numpy as np
>   
>   mesh = o3d.io.read_triangle_mesh(filename)
>   np_traingle = np.asarray(mesh.triangles)
>   
>   tri = {}
>   for i in range(np_traingle.shape[0]):
>       key = np_traingle[i][0]
>       tri.setdefault(key, set())
>       tri[key].add(np_traingle[i][1])
>       tri[key].add(np_traingle[i][2])
>       
>       key = np_traingle[i][1]
>       tri.setdefault(key, set())
>       tri[key].add(np_traingle[i][0])
>       tri[key].add(np_traingle[i][2])
>       
>       key = np_traingle[i][2]
>       tri.setdefault(key, set())
>       tri[key].add(np_traingle[i][0])
>       tri[key].add(np_traingle[i][1])
>   
>   tri = dict(sorted(tri.items()))
>   row1 = []
>   row2 =[]
>   
>   for key, value in tri.items():
>       tmp1 = [key for x in range(len(value))]
>       row1.extend(tmp1)
>       tmp2 = list(value)
>       row2.extend(sorted(tmp2))
>         
>   edge_index = [row1,row2]

But after executing this code edge_index shape comes [2, 2*x] (x is number of edges of in Mesh mentioned in off file)
I'm not sure wether I'm going in correct direction or not ?
Please let me know wether I'm correct or not and if not let me know how to convert mesh connectivity to edge_index.
Thanks in advance !!!
Following link contain one off file from my dataset
[MeshFile](https://drive.google.com/file/d/12ZYPvzTG6pp7iQONiQdAjDgtrckioKYk/view?usp=sharing)

**Comment by rusty1s** [comment, marked as answer]
This looks correct to me since PyG expects that both directions of an edge are present in `edge_index`. You can also take a look at the `FaceToEdge` transform for this.

**Reply by atale480** [reply]
Okay. Thanks for the help.
